pkg/otlp/config.go

```diff
@@ -163,12 +67,17 @@ func fromExperimentalConfig(cfg config.Config) (PipelineConfig, error) {
 
 // IsEnabled checks if OTLP pipeline is enabled in a given config.
 func IsEnabled(cfg config.Config) bool {
-	// TODO (AP-1267): Check stable config too
-	return isSetExperimental(cfg)
+	// HACK: We want to mark as enabled if the section is present, even if empty, so that we get errors
```

Comment:
Would it make sense to check `metrics_enabled` or `traces_enabled` instead? I don't think we enable anything by having an empty section in the config yet.

Reply:
I will let @mx-psi answer

Reply:
We want this to work exactly as it does on the OpenTelemetry Collector OTLP receiver (so that people can move easly from it to the Datadog Agent by copy-pasting, and so that we are sure that we are spec-compliant and provide all must-have flags). The OpenTelemetry Collector enables some stuff when sections are present but empty, so we do the same here.

The `otlp_config.metrics.enabled` and `otlp_config.traces.enabled` are there mostly for troubleshooting/having a workaround in case we ever have some serious bug which only happens with metrics/traces, but I wouldn't expect most users to use them. Maybe the problem here is that their names are confusing, and should be `disabled: false`?

Reply:
Also note that right now 
```
otlp_config:
   receiver:
```
[would fail](https://github.com/open-telemetry/opentelemetry-collector/blob/7cdf8c7f1cbcab89721ef8576ff0a5aab2d81f1e/receiver/otlpreceiver/config.go#L60) with `empty config for OTLP receiver`. The component would be enabled in this case (since the user has written the configuration), but invalid.

Reply:
Makes sense, thank you. I think it is a bit unfortunate that we add a new way to enable things to the configuration, at least at the top level. This is a very minor concern though, just wanted to make sure we're not missing some obviously better way of doing this.

> Maybe the problem here is that their names are confusing, and should be disabled: false?
No, this is fine. I missed that they were true by default.

Reply:
> I think it is a bit unfortunate that we add a new way to enable things to the configuration, at least at the top level

Yes, I agree that it is unfortunate that the configuration style on the OpenTelemetry Collector is different from the one on the Datadog Agent, but I think the trade-off is worth it since existing users of OpenTelemetry will be very familiar with the OpenTelemetry Collector (every other vendor has a distro of the Collector)